clang/test/dpct/query_api_mapping/cuDNN/dnn_part1.cu

```diff
@@ -94,3 +94,1287 @@
+// cudnnSoftmaxBackward-NEXT:   cudnnCreate(&h /*cudnnHandle_t **/);
+// cudnnSoftmaxBackward-NEXT:   cudnnSoftmaxBackward(h /*cudnnHandle_t*/, a /*cudnnSoftmaxAlgorithm_t*/,
+// cudnnSoftmaxBackward-NEXT:                        m /*cudnnSoftmaxMode_t*/, alpha /*void **/,
+// cudnnSoftmaxBackward-NEXT:                        dst_d /*cudnnTensorDescriptor_t*/, dst /*void **/,
+// cudnnSoftmaxBackward-NEXT:                        diff_dst_d /*cudnnTensorDescriptor_t*/,
+// cudnnSoftmaxBackward-NEXT:                        diff_dst /*void **/, beta /*void **/,
+// cudnnSoftmaxBackward-NEXT:                        diff_src_d /*cudnnTensorDescriptor_t*/,
+// cudnnSoftmaxBackward-NEXT:                        diff_src /*void **/);
+// cudnnSoftmaxBackward-NEXT: Is migrated to:
+// cudnnSoftmaxBackward-NEXT:   dpct::dnnl::engine_ext h;
+// cudnnSoftmaxBackward-NEXT:   h.create_engine();
+// cudnnSoftmaxBackward-NEXT:   h.async_softmax_backward(a, m, *alpha, dst_d, dst, diff_dst_d, diff_dst, *beta, diff_src_d, diff_src);
+
+// RUN: dpct --cuda-include-path="%cuda-path/include" --query-api-mapping=cudnnSoftmaxForward | FileCheck %s -check-prefix=cudnnSoftmaxForward
+// cudnnSoftmaxForward: CUDA API:
+// cudnnSoftmaxForward-NEXT:   cudnnHandle_t h;
+// cudnnSoftmaxForward-NEXT:   cudnnCreate(&h /*cudnnHandle_t **/);
+// cudnnSoftmaxForward-NEXT:   cudnnSoftmaxForward(h /*cudnnHandle_t*/, a /*cudnnSoftmaxAlgorithm_t*/,
+// cudnnSoftmaxForward-NEXT:                       m /*cudnnSoftmaxMode_t*/, alpha /*void **/,
+// cudnnSoftmaxForward-NEXT:                       src_d /*cudnnTensorDescriptor_t*/, src /*void **/,
+// cudnnSoftmaxForward-NEXT:                       beta /*void **/, dst_d /*cudnnTensorDescriptor_t*/,
+// cudnnSoftmaxForward-NEXT:                       dst /*void **/);
+// cudnnSoftmaxForward-NEXT: Is migrated to:
+// cudnnSoftmaxForward-NEXT:   dpct::dnnl::engine_ext h;
+// cudnnSoftmaxForward-NEXT:   h.create_engine();
+// cudnnSoftmaxForward-NEXT:   h.async_softmax_forward(a, m, *alpha, src_d, src, *beta, dst_d, dst);
+
+// RUN: dpct --cuda-include-path="%cuda-path/include" --query-api-mapping=cudnnTransformTensor | FileCheck %s -check-prefix=cudnnTransformTensor
+// cudnnTransformTensor: CUDA API:
+// cudnnTransformTensor-NEXT:   cudnnHandle_t h;
+// cudnnTransformTensor-NEXT:   cudnnCreate(&h /*cudnnHandle_t **/);
+// cudnnTransformTensor-NEXT:   cudnnTransformTensor(h /*cudnnHandle_t*/, alpha /*void **/,
+// cudnnTransformTensor-NEXT:                        src_d /*cudnnTensorDescriptor_t*/, src /*void **/,
+// cudnnTransformTensor-NEXT:                        beta /*void **/, dst_d /*cudnnTensorDescriptor_t*/,
+// cudnnTransformTensor-NEXT:                        dst /*void **/);
+// cudnnTransformTensor-NEXT: Is migrated to:
+// cudnnTransformTensor-NEXT:   dpct::dnnl::engine_ext h;
+// cudnnTransformTensor-NEXT:   h.create_engine();
+// cudnnTransformTensor-NEXT:   h.async_reorder(*alpha, src_d, src, *beta, dst_d, dst);
```

Comment:
Please observe the lit CI test time. If it is too long, need split this lit file to several files (may refer the CUDA doc's chapter).